app/src/test/java/com/nextcloud/talk/utils/TextMatchersTest.java

```diff
@@ -0,0 +1,89 @@
+package com.nextcloud.talk.utils;
+
+import com.nextcloud.talk.models.json.chat.ChatMessage;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class TextMatchersTest {
+
+    @Test
+    public void getMessageTypeFromString_assertRegularTextMessageParsing() {
```

Comment:
Could we name things differently, like "shouldBeRegularTextMessage" in this case? :)

Reply:
@mario hmm, why? :)

Reply:
Long method names make little sense, if method name changes you need to change the tests too (which can't be done by automatic refactoring).

In addition, using something like "shouldBeRegularTextMessage" (which is DEFINITELY not the only way to do it) makes it easy to read and you instantly know what's happening and what is expected.

One of the alternatives (that I personally dislike, again, because it's a long method name for very little gain), as an example:

classifyMessage_regularTextGiven_shouldClassifyAsRegularTextMessage

Note the pattern:

{functionality tested}_{input given}_{result expected}

Btw. thank you for the contributions in general, the community and me really appreciate it.

Reply:
There are different school of thoughts on this obviously. We both agree unit test method names should make it very clear (at the very least) what is being tested and the expected result.

I realize the downside of having test names tied to the method being tested but how often do you actually change method names? I've grown accustomed to using it because that's what Android Studio does automatically when you generate the test class. Also, most (but not all) of Android's developer documentation and training material on unit testing does the same.

That said, if you would like to enforce a different naming convention, I'm all for it.

Reply:
Obviously, which is why I said my opinion is not perfect nor one that solves all issues in the world :)

Which one of my two proposals above would you be more comfortable with? I understand the pain of having to change the auto-generated names by AS, but we also have to think about the future. :)

(btw. no matter which one you choose, feel free to add a WRITING TESTS document where we outline our decision).

As for method names - not that often, but they'll probably be changed as we refactor the code since the current state is "young and naive" :)

Reply:
check out the updated test names. comment?